src/applications/gi/selectors/calculator.js

```diff
@@ -451,6 +451,28 @@ const getDerivedValues = createSelector(
 
     const totalHousingAllowance = monthlyRateFinal * termLength;
 
+    let bah;
+    // if beneficiary has indicated they are using a localized rate and beneficiaryLocationBah exists, then a localized rate has been fetched and should be used
+    const useBeneficiaryLocationRate = inputs.beneficiaryLocationQuestion === 'no' &&
+      inputs.beneficiaryLocationBah !== null;
+    // if beneficiary has indicated they are using the grandfathered rate, use it when available;
+    const useGrandfatheredBeneficiaryLocationRate = inputs.giBillBenefit === 'yes';
```

Comment:
Added some comments and tried to use some meaningful names here. 

- if the user has indicated that they started using their GI Bill benefit before 1/1/18, then we use the grandfathered rate if it's available (sometimes it's not).